### ISSUE TITLE
Not working with PSTV

### ISSUE THREAD

**Issue by orangelampshade** [CallmeBK/VolLightbarCtrlWithDS4#2]
Hello. Fantastic plug-in, I love the LED colour changing feature (reason I installed). However I did notice that it doesn’t seem to work on my PSTV running 3.60 enso. Now I know it’s described as being for the Vita, but if you ever get around to it, it would be nice if it worked with the PSTV (in case you didn’t know). I appreciate your effort.

**Comment by CallmeBK** [comment]
Oh, I thought it would work for pstv 3.60 henkaku. I've never been able to
test for pstv though because I don't own one. First, I want to know if you
have a vita on 3.60 and got it working for that?

**Comment by orangelampshade** [comment]
I don’t have a vita but I imagine it works. In theory the pstv should work, it’s the same hardware. I can trigger the volume bar, it just won’t go up or down. Strange...

**Comment by CallmeBK** [comment]
Sorry for taking so long to look into this, but I think I have an idea as to why this is happening and how I can fix it.  If you don't have reVita plugin installed on your psTV, then install it.  Then open the menu by pressing start + square.  Go to profile > Controller and upload a picture of that screen to this chat so I can see it.

**Comment by CallmeBK** [comment]
![IMG_20210209_080934](https://user-images.githubusercontent.com/66493237/107376832-f5c3c100-6aaf-11eb-9abd-c5ac9bef3848.jpg)

**Comment by CallmeBK** [comment]
This is what it looks like on a PS vita with no controllers connected. port 0 is always reserved for the PS vita itself so my code actually only checks ports 1 through 4. I'm thinking that your controller was on port 0 because you are on pstv so I just need to update my code to check port 0 through 4. It totally slipped my mind that the ds4 controllers on pstv could be using port 0.

But I would like you to upload the image I requested first so I can confirm my suspicions

**Comment by orangelampshade** [comment]
That’s fine, I know this is not a priority. I followed your instructions but it looks like it’s using port 1. Even if your theory was correct, one would think the volume bar couldn’t be triggered at all. Like I said, it can be triggered, but i can’t make the volume go up or down. Thanks

![7B84776F-5748-470C-B45E-3A30E3275593](https://user-images.githubusercontent.com/58881250/107466009-7f11eb00-6bb7-11eb-8e0c-5f21724f226f.jpeg)

**Comment by CallmeBK** [comment]
Hmm yeah that was the only doubt I had with my theory.  Does the same thing happen in these scenarios:

1) Pressing share + L1
2) Pressing share + R1
3) Holding share + L1 + R1

And then if you hold the buttons down in scenario 1) and 2), does the volume bar stay on the screen until you let go?

And did you try any of this with a different DS4 controller?

**Comment by orangelampshade** [comment]
I tried everything and it behaves as expected. Volume bar is triggered but will not move no matter the combination of buttons I press. If I were to hold the trigger buttons the volume bar remains on the screen until I let go. I did use a different ds4, yes. The problem persists. Have you tried it with a ds4 connected to the vita? Thanks 

![A57F3925-C0E1-40E9-8761-F94138121BF6](https://user-images.githubusercontent.com/58881250/107611597-76451600-6c98-11eb-8310-82a81a3e5695.jpeg)

**Comment by orangelampshade** [comment]
![Uploading 950C8052-045E-463D-8C37-01AB3DA237B7.jpeg…]()

**Comment by CallmeBK** [comment]
Yeah here's a demonstration of how it is supposed to work.

https://user-images.githubusercontent.com/66493237/107642167-20418580-6c3a-11eb-88ab-7639e88b5aec.mp4

Is the lightbar portion working then?  Just the volume control that doesn't?

What does your config.txt look like?  Maybe there's another plugin that conflicts with this one.

**Comment by orangelampshade** [comment]
![AF96D507-062C-4C94-9A57-5D993E89109B](https://user-images.githubusercontent.com/58881250/107833355-1a3cd780-6de7-11eb-966e-bceb6bea545a.jpeg)
![77023F13-8BA7-45A0-B88A-B2285522A79D](https://user-images.githubusercontent.com/58881250/107833359-1f018b80-6de7-11eb-8d7f-f26facac9bbe.jpeg)

I don’t have a great deal oh plug-ins installed, maybe you can tell better than me if something is wrong in my config file. 

Lightbar feature works great. I absolutely love it! It would be nice to know if there are other users out there with a PSTV experiencing the same problem.

**Comment by CallmeBK** [comment]
Hmm I have a lot of those plugins on my vita, but not all of them. To eliminate the possibility of a conflicting plugin, make a backup of your config.txt and then try disabling all other plugins except for mine.

In the meantime, I'm asking if anyone has used my plugin on PSTV in a couple of Discord channels.

Oh and to help you understand the config.txt file a bit better, here is a good reference that helped clear my confusion when I first started:

https://tai.henkaku.xyz/

Scroll down to the configuration section

**Comment by orangelampshade** [comment]
Appreciate the info. Makes a bit more sense now. It will be valuable to know if others are experiencing the same problem, or if it’s just me. If you find out please let me know. In the meantime I’ll mess around with the plug-ins. Thanks.

**Comment by CallmeBK** [comment]
I haven't heard from any other pstv users that have used my plugin. I do have another theory though. Try to use the reVita plugin to map a button to the volume+ button. Then try pressing the button you mapped to see if the volume increases or if it behaves exactly like my plugin where the volume bar shows on screen but doesn't do anything.

If the volume does indeed increase, then you can use the reVita plugin to do what mine is supposed to do. 

If the volume does not increase and behaves like mine, then I have an idea for updating my code to get it to work on psTV.

**Comment by CallmeBK** [comment]
I was able to get feedback from another PSTV user. They're having the same issue:

https://www.reddit.com/r/VitaPiracy/comments/lju28a/vollightbarctrlwithds4_pstv_feedback_needed/

**Comment by orangelampshade** [comment]
Awesome I saw that you got what you needed in terms of testing. What is your idea for the fix or shall I just wait lol?

**Comment by CallmeBK** [comment]
I know of a function that can adjust the master volume from the quick menu. For PSTV users, I'll trigger that function instead of volume button presses. The only issue is that no volume indication will show up on the screen when the master volume is changed.

I'll probably try to release it sometime this week. Then I'll work on figuring out how to draw my own volume indicator bar. That part will take more time.

**Comment by orangelampshade** [comment]
Makes sense. You can’t borrow the code that displays the ps volume bar?

**Comment by CallmeBK** [comment]
When I was looking for that code, I couldn't find it. But I did find this plugin made by cuevavirus that brings the PS volume bar onto the screen:

https://forum.devchroma.nl/index.php/topic,46.0.html

I would try to see if this works for pstv. If it doesn't, I would be totally lost trying to get the PS volume bar to show up and wouldn't know where to look.

**Comment by orangelampshade** [comment]
To be honest it’s not a big deal. The light function was what I wanted anyway, the volume bar was just a bonus. Adjusting the volume without an indicator is still handy though. See what you can do.

**Comment by CallmeBK** [comment]
At the very least I'll get the volume control working on the PSTV. And then some sort of on screen indicator.

Getting the indicator to look nicer or exactly like the built in PS volume indicator I'll try to figure out eventually, but not sure when I'll get around to it

**Comment by orangelampshade** [comment]
I wish I could help lol. Will you post here when you get around to it so I can update it? Thanks

**Comment by CallmeBK** [comment]
Haha it's all good. You've been a big help just being there to give me feedback because I don't have a psTV myself. I really appreciate it and want to provide you the full experience that I had intended with this plugin.

**Comment by orangelampshade** [comment]
All good man I’ll wait to hear back. Thanks

**Comment by CallmeBK** [comment]
I think I've got something that would work, but I want you to try it for me first before I upload it as the next release.  What's your email so I can send the files/instructions directly to you?

**Comment by orangelampshade** [comment]
I sent you an email so just reply to that.

**Comment by v1ru555** [comment]
> I think I've got something that would work, but I want you to try it for me first before I upload it as the next release. What's your email so I can send the files/instructions directly to you?

If you need additional help again (i was the one on Reddit), I could revive my PSTV again to test.

**Comment by CallmeBK** [comment]
I've been going back and forth with orangelampshade over email so I could send the plugin files as attachments for him to test. We found out that volume control is disabled on PSTV and I'm currently looking into how to enable it. In the meantime, I also found a function in the software for changing the master volume that hasn't been reverse engineered yet by the homebrew community. Hopefully that one will work on the PSTV because the function name matches the "master volume" slider in the quick menu. I'll probably release that method first so you guys at least have something that works soon as I'm not sure how long it will take me to figure out how to enable volume control on PSTV
